src/components/image/image.test.tsx

```diff
@@ -0,0 +1,127 @@
+
+it("renders", () => {
+  render(<Image data-role="image" />);
+
+  expect(screen.getByTestId("image")).toBeInTheDocument();
+});
+
+it("renders with the 'src' prop", () => {
+  render(
+    <Image
+      data-role="image"
+      src="https://upload.wikimedia.org/wikipedia/en/6/6c/Heatposter.jpg"
+      alt="Heat poster"
+    />
+  );
+
+  expect(screen.getByRole("img")).toBeInTheDocument();
+});
+
+it("renders with the 'src' and 'decorative' props", () => {
+  render(
+    <Image
+      data-role="image"
+      src="https://upload.wikimedia.org/wikipedia/en/6/6c/Heatposter.jpg"
+      alt=""
+      decorative
+    />
+  );
+
+  expect(screen.getByRole("img")).toBeInTheDocument();
+});
+
+it("renders with children", () => {
+  render(<Image>Children</Image>);
+
+  const children = screen.getByText("Children");
+  expect(children).toBeInTheDocument();
+});
+
+it("does not render, as an invariant is fired due to children being passed as well as the 'src' prop", () => {
+  const consoleSpy = jest.spyOn(console, "error").mockImplementation(() => {});
+
+  expect(() =>
+    render(
+      <Image
+        src="https://upload.wikimedia.org/wikipedia/en/6/6c/Heatposter.jpg"
+        alt="Heat poster"
+      >
+        Children
+      </Image>
+    )
+  ).toThrow(
+    "The 'Image' component renders as an 'img' element when the 'src' prop is used and therefore does not accept children."
+  );
+
+  consoleSpy.mockRestore();
+});
+
+it("does not render, as an invariant is fired due to the 'src' prop being passed without an 'alt' prop", () => {
+  const consoleSpy = jest.spyOn(console, "error").mockImplementation(() => {});
+
+  expect(() =>
+    render(
+      <Image src="https://upload.wikimedia.org/wikipedia/en/6/6c/Heatposter.jpg" />
+    )
+  ).toThrow(
+    "Please use the 'decorative' prop if the 'alt' text should be an empty value or provide an 'alt' string when rendering the 'Image' component as an 'img' element."
+  );
+
+  consoleSpy.mockRestore();
+});
+
+it("does not render, as an invariant is fired due to the 'src' prop being passed and an empty string being passed to the 'alt' prop (the decorative prop must be passed)", () => {
+  const consoleSpy = jest.spyOn(console, "error").mockImplementation(() => {});
+
+  expect(() =>
+    render(
+      <Image
+        src="https://upload.wikimedia.org/wikipedia/en/6/6c/Heatposter.jpg"
+        alt=""
+      />
+    )
+  ).toThrow(
+    "Please use the 'decorative' prop if the 'alt' text should be an empty value or provide an 'alt' string when rendering the 'Image' component as an 'img' element."
+  );
+
+  consoleSpy.mockRestore();
+});
+
+it("sets the 'alt' attribute correctly when a custom value is passed to the 'alt' prop", () => {
+  render(
+    <Image
+      data-role="image"
+      src="https://upload.wikimedia.org/wikipedia/en/6/6c/Heatposter.jpg"
+      alt="Heat poster"
+    />
+  );
+
+  expect(screen.getByRole("img")).toHaveAttribute("alt", "Heat poster");
```

Comment:
**suggestion:** I think you could use an accessible name assertion for the alt text.  
```suggestion
  expect(screen.getByRole("img")).toHaveAccessibleName("Heat poster");
```